lib/ansible/module_utils/powershell/Ansible.ModuleUtils.PrivilegeUtil.psm1

```diff
@@ -0,0 +1,524 @@
+    $name = The name of the privilege, this is the constant value from https://docs.microsoft.com/en-us/windows/desktop/SecAuthZ/privilege-constants, e.g. SeAuditPrivilege
+    $new_state = 'System.Collections.Generic.Dictionary`2[[System.String], [System.Nullable`1[System.Boolean]]]'
+        The key is the constant name as a string, the value is a ternary boolean where
+            true - will enable the privilege
+            false - will disable the privilege
+            null - will remove the privilege
+
+    Each method that changes the privilege state will return a dictionary that
+    can be used as the $new_state arg of SetTokenPrivileges to undo and revert
+    back to the original state. If you remove a privilege then this is
+    irreversible and won't be part of the returned dict
+    #>
+    [CmdletBinding()]
+    # build the C# code to compile
+    $namespace_import = ($ansible_privilege_util_namespaces | ForEach-Object { "using $_;" }) -join "`r`n"
+    $platform_util = "$namespace_import`r`n`r`n$ansible_privilege_util_code"
+
+    # FUTURE: find a better way to get the _ansible_remote_tmp variable
+    # this is used to force csc to compile the C# code in the remote tmp
+    # specified
+    $original_tmp = $env:TMP
+
+    $remote_tmp = $original_tmp
+    $module_params = Get-Variable -Name complex_args -ErrorAction SilentlyContinue
+    if ($module_params) {
+        if ($module_params.Value.ContainsKey("_ansible_remote_tmp") ) {
+            $remote_tmp = $module_params.Value["_ansible_remote_tmp"]
+            $remote_tmp = [System.Environment]::ExpandEnvironmentVariables($remote_tmp)
+        }
+    }
+
+    $env:TMP = $remote_tmp
+    Add-Type -TypeDefinition $platform_util
+    $env:TMP = $original_tmp
+}
+
+Function Get-AnsiblePrivilege {
+    <#
+    .SYNOPSIS
+    Get the status of a privilege for the current process. This returns
+        $true - the privilege is eanbled
+        $false - the privilege is disabled
+        $null - the privilege is removed from the token
+
+    If Name is not a valid privilege name, this will throw an
+    ArgumentException.
+
+    .EXAMPLE
+    Get-AnsiblePrivilege -Name SeDebugPrivilege
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)][String]$Name
+    )
+
+    if (-not [Ansible.PrivilegeUtil.Privileges]::CheckPrivilegeName($Name)) {
+        throw [System.ArgumentException] "Invalid privilege name '$Name'"
+    }
+
+    $process_token = [Ansible.PrivilegeUtil.Privileges]::GetCurrentProcess()
+    $privilege_info = [Ansible.PrivilegeUtil.Privileges]::GetAllPrivilegeInfo($process_token)
+    if ($privilege_info.ContainsKey($Name)) {
+        $status = $privilege_info.$Name
+        return $status.HasFlag([Ansible.PrivilegeUtil.PrivilegeAttributes]::Enabled)
+    } else {
+        return $null
+    }
+}
+
+Function Remove-AnsiblePrivilege {
```

Comment:
I assume the Remove stuff is just here for completeness' sake, but anyone who actually used it will cause problems with persistent interpreters... Might want to skip including this just as an "attractive nuisance" unless we actually need it for something.

Reply:
Yea, really completeness. I honestly don't see a use for this in anything we do today. Sounds good to remove this cmdlet. Someone can call `RemovePrivilege` or `SetTokenPrivileges` if they really wanted to do it.